### ISSUE TITLE
Campaign Monitor - ConsentToTrack is set to Yes but still Failed Queue

### ISSUE THREAD

**Issue by pbmills** [verbb/formie#2076]
### Describe the bug

I am getting this error
<img width="1720" alt="Screenshot 2024-10-02 at 2 41 38 PM" src="https://github.com/user-attachments/assets/ebe1e266-d5c4-445c-9be8-559dd443844e">

In the Integration I have set the ConsentToTrack 'Yes'
<img width="1458" alt="Screenshot 2024-10-02 at 2 40 53 PM" src="https://github.com/user-attachments/assets/18fd652a-2fd6-484a-967f-1849a6aa8434">

### Steps to reproduce

1.  Set the ConsentToTrack to Yes
2. Fill the form and send a form

### Form settings

- Multi-page form: No
- Submission Method: Ajax 

### Craft CMS version

5.4.5.1

### Plugin version

3.0.7

### Multi-site?

No

**Comment by engram-design** [comment]
Can't seem to replicate this sorry. Can you confirm if you've refreshed the integration (clicking the refresh button next to where you select your list)? I can see your mappings to consent values are set to "Yes", so that otherwise looks correct to me.

**Comment by pbmills** [comment]
Yes, I have clicked on the refresh button next to where the select list is.

The job data shows null:

"ConsentToTrack": "",
________________________________

**Comment by engram-design** [comment]
Hmm, a blank value should be impossible, as `Yes` or `No` are the only two options [we set](https://github.com/verbb/formie/blob/12905b31c62eb964e2e06d006b6ce6eccf3f64ed/src/integrations/emailmarketing/CampaignMonitor.php#L163).

There's no custom module code or anything altering behaviour?

**Comment by pbmills** [comment]
No custom module code.  Just a basic form which has Name, Email, privacy policy checkbox.



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by engram-design** [comment]
Can you send through a support request via Formie > Settings > Get Support for that form? I would like to see your integration mappings as stored in the database, just to double check things are all okay.

**Comment by pbmills** [comment]
thank you. I have just send Support Request from the Formie settings.

________________________________

**Comment by engram-design** [comment]
Thanks for that, I can confirm your form settings are all correct. As an example, my job detail looks like:

```
{
    "description": null,
    "submissionId": 16796,
    "payload": {
        "EmailAddress": "support@verbb.io",
        "Name": null,
        "CustomFields": [],
        "Resubscribe": true,
        "RestartSubscriptionBasedAutoresponders": true,
        "ConsentToTrack": "Yes",
        "ConsentToSendSms": "No"
    },
    "integration": {
        "apiKey": ".........",
        "clientId": ".........",
        "fieldMapping": {
            "Email": "{field:email}",
            "Name": "{field:multiName.firstName}",
            "MobileNumber": "",
            "ConsentToTrack": "true",
            "ConsentToSendSms": "false"
        },
        "listId": "b26198236334d2a12d6f8f92c45b2",
```

**Comment by pbmills** [comment]
Thank you for taking a look at it. What would you think the issue could be at my side?  Should I need to do reintegration? I am trying to re-run the failed queue and it could be the failed queue isn't storing the ConsentToTrack value when I re-run it?
________________________________

**Comment by engram-design** [comment]
What does the job detail look like compared to my above example? Re-running the job should update any field mapping you have defined in your form settings.
